doc/ApiDocGeneration/Generate-Api-Docs.ps1

```diff
@@ -0,0 +1,110 @@
+# Generates API Docs from the Azure-SDK-for-NET repo
+ 
+[CmdletBinding()]
+Param (
+    $ArtifactName,
+    $ServiceDirectory,
+    $ArtifactsDirectoryName,
+    $LibType,
+    $RepoRoot,
+    $BinDirectory
+)
+
+Write-Verbose "Create variables for identifying package location and package safe names"
+$PackageLocation = "$($ServiceDirectory)/$($ArtifactName)"
+Write-Verbose "Package Location $($PackageLocation)"
+
+if ($ServiceDirectory -eq '*') {
+    $PackageLocation = "core/$($ArtifactName)"
+}
+
+if ($ServiceDirectory -eq 'cognitiveservices') {
+    $PackageLocation = "cognitiveservices/$($ArtifactsDirectoryName)"
+}
+
+if ($LibType -eq 'Management') {
+    $ArtifactName = $ArtifactName.Substring($ArtifactName.LastIndexOf('.Management') + 1)
+}
+
+#Write-Verbose "Set variable for publish pipeline step"
+#echo "##vso[task.setvariable variable=artifactsafename]$($SafeName)"
+
+Write-Verbose "Create Directories Required for Doc Generation"
+mkdir "$($BinDirectory)/$($ArtifactName)/dll-docs/my-api"
+mkdir "$($BinDirectory)/$($ArtifactName)/dll-docs/dependencies/my-api"
+mkdir "$($BinDirectory)/$($ArtifactName)/dll-xml-output"
+mkdir "$($BinDirectory)/$($ArtifactName)/dll-yaml-output"
+mkdir "$($BinDirectory)/$($ArtifactName)/docfx-output"
+
+Write-Verbose "Name Reccuring paths with variable names"
+$FrameworkDir = "$($BinDirectory)/$($ArtifactName)/dll-docs"
```

Comment:
I'd move these above the mkdir section and let the mkdir calls use them as well.